alacritty_terminal/src/term/mod.rs

```diff
@@ -1190,14 +1228,14 @@ impl<T> Term<T> {
         &self.cursor
     }
 
-    pub fn swap_alt(&mut self) {
-        if self.alt {
+    pub fn swap_grids(&mut self) {
+        if self.swapped {
             let template = self.cursor.template;
             self.grid.region_mut(..).each(|c| c.reset(&template));
         }
 
-        self.alt = !self.alt;
-        mem::swap(&mut self.grid, &mut self.alt_grid);
+        self.swapped = !self.swapped;
+        mem::swap(&mut self.grid, &mut self.inactive_grid);
     }
```

Comment:
We should move everything related to alt screen swapping into this method, that would make using it a lot easier.

So this should insert/remove the term mode, save/restore cursor position and swap the screens.

If you feel like doing a little bit more work, @kchibisov and myself have also come to the conclusion that there is no need for three cursor states (cursor, cursor_save and cursor_save_alt). Instead we should be able to do the same thing we do with the grid and just `mem::swap` them out, so that we have a `cursor` and `inactive_cursor` and nothing else. Then we'd never have to check for the mode to determine which save cursor we want to use. The naming would also be more consistent.

Reply:
I don't mind trying out a bit of a refactor with all of this 👍